README.md

```diff
@@ -1,13 +1,135 @@
+$ cd kube-exec-controller
+$ make deploy
+```
+
+To see how kube-exec-controller works, let's create a test Pod in your local cluster and send a `kubectl exec` request to it:
+```
+$ kubectl run test --image=nginx
+pod/test created
+
+$ kubectl exec test -- touch new-file
+```
+
+You will see the test Pod has some labels attached and receives corresponding K8s events from our controller app:
+```
+$ kubectl get pod --show-labels
+NAME   READY   STATUS    RESTARTS   AGE   LABELS
+test   1/1     Running   0          8s    box.com/podInitialInteractionTimestamp=1634408037,box.com/podInteractorUsername=kubernetes-admin,box.com/podTTLDuration=2m0s,run=test
+
+$ kubectl describe pod test
+...
+Warning  PodInteraction  68s   kube-exec-controller  Pod was interacted with 'kubectl exec/attach' command by a user 'kubernetes-admin' initially at time 2021-10-16 18:04:44.5257517 +0000 UTC m=+27.185038701
+Warning  PodInteraction  68s   kube-exec-controller  Pod will be evicted at time 2021-10-16 18:06:44 +0000 UTC (in about 1m59s)
+```
+
+You can also utilize the `kubectl pi` plugin to get more detailed info or request an extension to the test Pod's eviction time:
+```
+$ kubectl pi get
+POD_NAME  INTERACTOR        POD_TTL  EXTENSION  EXTENSION_REQUESTER  EVICTION_TIME
+test      kubernetes-admin  2m0s                                     2021-10-16 18:06:44 +0000 UTC
+
+$ kubectl pi extend --duration=1m
+Successfully extended the termination time of pod/test with a duration=1m
+
+$ kubectl pi get
+POD_NAME  INTERACTOR        POD_TTL  EXTENSION  EXTENSION_REQUESTER  EVICTION_TIME
+test      kubernetes-admin  2m0s     1m         kubernetes-admin     2021-10-16 18:07:44 +0000 UTC
+
+$ kubectl describe pod test
+...
+Warning  PodInteraction  29s   kube-exec-controller  Pod eviction time has been extended by '1m', as requested from user 'kubernetes-admin'. New eviction time: 2021-10-16 18:07:44 +0000 UTC
+Warning  PodInteraction  29s   kube-exec-controller  Pod will be evicted at time 2021-10-16 18:07:44 +0000 UTC (in about 2m21s)
+```
+
+## Usage
+#### kube-exec-controller
+```
+$ kube-exec-controller --help
+Usage of /kube-exec-controller:
+  -api-server string
+    	URL to K8s api-server, required if kube-proxy is not set up
+  -cert-path string
+    	Path to the PEM-encoded TLS certificate
+  -extend-chan-size int
+    	Buffer size of the channel for handling Pod extension (default 500)
+  -interact-chan-size int
+    	Buffer size of the channel for handling Pod interaction (default 500)
+  -key-path string
+    	Path to the un-encrypted TLS key
+  -log-level debug
+    	Log level. debug, `info`, `warn`, `error` are currently supported (default "info")
+  -namespace-allowlist string
+    	Comma separated list of namespaces that allow interaction without evicting their Pods
+  -port int
+    	Port for the app to listen on (default 8443)
+  -ttl-seconds int
+    	TTL (time-to-live) of interacted Pods before getting evicted by the controller (default -1)
```

Comment:
It looks like the default is set to -1 as a way to force the user to provide a value. Instead of that, I would recommend setting default to a low value, say 10 mins, which will allow users to get up and running quickly and advanced users the ability to override as needed.